plugins/outputs/sql/sql.go

```diff
@@ -309,7 +316,15 @@ func (p *SQL) Write(metrics []telegraf.Metric) error {
 
 // Convert a DSN possibly using v1 parameters to clickhouse-go v2 format
 func (p *SQL) convertClickHouseDsn() {
-	u, err := url.Parse(p.DataSourceName)
+	buf, err := p.DataSourceName.Get()
+	if err != nil {
+		p.Log.Errorf("loading data source name failed: %v", err)
+		return
+	}
+	dsn := buf.String()
+	buf.Destroy()
+
+	u, err := url.Parse(dsn)
```

Comment:
```suggestion
	defer buf.Destroy()

	u, err := url.Parse(buf.String())
```
Maybe also rename `buf` to `dataSourceName`?

Reply:
Please leave the code as is. Using `defer` will cause us to hold the locked memory until the connection completes, which in the worst case will be a user-configured timeout!!!! As most systems are configured with a low limit for locked pages this will cause trouble!

Reply:
Yes, see previous conversation, so what about using like this? That way the raw value is not in memory scope of this method.

```suggestion
	u, err := url.Parse(buf.String())
	buf.Destroy()
```

Reply:
I don't understand what we gain by this. Can you explain more?

Reply:
During the whole execution of `convertClickHouseDsn` the raw DSN is in memory as plain text (`dsn` var) instead of only during `url.Parse`.

Reply:
That's about the other thread, please reply over there..